[DOC] Create new doc for Dashboard application (#2350)
opensearch-project/documentation-website: _dashboards/dashboard/index.md

Before getting started, let's get familiar with the **Dashboard** user interface. The user interface comprises the following main parts:

![Dashboard user interface]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-UI.png)

- The **navigation panel** (A) on the left contains the OpenSearch Dashboards applications.
- The **search** bar (B) lets you search for documents and other objects and add filters.
- The **filter** (C) lets you apply different data perspectives to narrow a dashboard's results.
- The **toolbar** (D) contains frequently used commands and shortcuts.
- The **time filter** (E) lets you customize the time and date.
- The **panel** (F) allows you to add existing visualizations to the dashboard or create new ones for the dashboard.

## Definitions

Learn some useful terminology for working with OpenSearch Dashboards and the **Dashboard** application.

- _Dashboards_ is the abbreviated proper name for OpenSearch Dashboards. OpenSearch Dashboards is an open-source visualization tool designed to work with OpenSearch.
- _Dashboard_ is the application in OpenSearch Dashboards that is used to track, analyze, and display data at a glance.

> - _dashboard_ or _dashboards_ refer to common names for a tool used to visually display data.

Reviewer: should this be "refers" ?

Reviewer 2: how about _dashboard_ and _dashboards_ are common names for the tools used...

> - _Panel_ is a term used to refer to a visualization displayed on a dashboard. The terms _panel_ and _visualization_ may be used interchangeably throughout this and other Dashboard-related documentation.

Reviewer 3: Helpful! ❤️

1. From the panel, choose **Add an existing**.
1. From the **Add panels** window, choose **[eCommerce] Promotion Tracking**, and then select `x` to close the panel.

You've now created the following basic dashboard with a single panel, which you'll continue using throughout this tutorial.

![Basic dashboard with single panel]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-basic.png)

## Creating visualizations

Continuing with the dashboard you created in the preceding steps, you'll create a new visualization and save it to the dashboard.

1. From the dashboard toolbar, choose **Create new**, then **Visualization**.
1. From the **New Visualization** window, choose **Vertical Bar** and then select the index pattern **opensearch_dashboards_sample_data_ecommerce**. 
1. From the toolbar, select **Save**
1. In the **Save visualization** window, add a title for the visualization. For example, the title for the gauge chart panel is [eCommerce] Orders.
1.  Select **Save and return**.  

The gauge chart visualization is now saved and you are taken back to the dashboard. You'll see two visualizations on the dashboard, like the following: 

> ![Dashboard showing visualizations combined in a single view]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-combined.png)

Reviewer 3: These steps and the screenshot refer to a gauge chart, but L62 specifies "Vertical Bar", and should be updated to match.

## Adding subsequent panels

Continuing with the dashboard you created in the preceding steps, you'll add a visualization to the dashboard.

1. From the dashboard toolbar, choose **Add**.
1. From the **Add panels** window, choose **[eCommerce] Sales by Category**.
1. Select `x` to close the **Add panels** window. 

You'll see an area chart visualization display on the dashboard, as shown in the following image: 

![Adding another panel to the dashboard]({{site.url}}{{site.baseurl}}/images/dashboards/new-area-viz.png)

## Saving dashboards

When you've finalized your dashboard, save it. If you're saving a new dashboard:

1. In the toolbar, choose **Save**.
2. On the **Save dashboard** window, enter the **Title**. The **Description** is optional.
3. To save the time filter to the dashboard, select **Store time with dashboard**.
4. Choose **Save**.

## Customizing the look of a panel

To customize the panels, you'll need to be in edit mode:

1. Select **Edit** at top right in the toolbar. 

Showing a legend can give readers more information, while hiding a legend can give it a cleaner look. If you want to show or hide the panel legend:

> 1. To show the legend, 

Reviewer: how do I show the legend?

Author: Added that instruction. Good catch.

1. Select the list icon in the panel's lower left corner.

If you want to change the color of the panel legend:

1. Select the gear icon on the area chart panel.
2. From the **Options** window, select **Edit visualization**.
3. From the visualization legend, select a category and then select a color from the flyout. The area chart updates with your change.

> 4. Select **Save and return**. 

Reviewer 3: Note that it _is_ possible to change chart colors directly from the legend within a dashboard panel. The only difference is that those changes are only saved for that particular dashboard and panel, and don't affect the saved visualization. These steps save the colors on the visualization itself, which will apply to any dashboard that links to the visualization (unless they have already specified different panel colors).

> If you want to show or hide panel title or customize it without applying the changes across the dashboard overall:

Reviewer 3: "without applying the changes across the dashboard overall" isn't quite right - changing panel titles only affects the particular panel on the particular dashboard, and won't affect any other panel containing that same visualization (or any other dashboard).

1. Select the gear icon on the panel.
2. From the **Options** window, select **Edit panel title**.
3. From the **Customize panel**, add a title under **Panel title** or toggle the **Show panel title** to hide the title.
4. Choose **Save**.

## Arranging panels

To organize panels, arrange them side-by-side, or resize them, you can use these options:

- To move, select and hold the panel title and drag to the new location.
- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.

> - To view in fullscreen mode, select the gear icon (edit mode) or vertical ellipsis () in the top right of the panel and select **Maximize panel**. To minimize the fullscreen mode, select the vertical ellipsis and select **Minimize**.

Reviewer: should the ellipsis be in the parenthesis?

Author: Added

{: .note}

Showing a legend can give readers more information, while hiding a legend can give it a cleaner look. If you want to show or hide the panel legend:

- Select the list icon in the panel's lower left corner.

If you want to change the color of the panel legend:

- From the visualization legend, select a category and then select a color from the flyout. The area chart updates with your change.

This color change is only saved for the current panel and dashboard and doesn't affect the saved visualization.
{: .note}

If you want to change the color of the panel legend in the visualization:

1. Select the gear icon on the area chart panel.
2. From the **Options** window, select **Edit visualization**.
3. From the visualization legend, select a category and then select a color from the flyout. The area chart updates with your change.
4. Select **Save and return**. 

This color change affects the saved visualization and any dashboard that links to the visualization.
{: .note}

If you want to show or hide panel title or customize it:

1. Select the gear icon on the panel.
2. From the **Options** window, select **Edit panel title**.

> 3. From the **Customize panel**, add a title under **Panel title** or toggle the **Show panel title** to hide the title.

Reviewer 4: ```suggestion
3. From the **Customize panel**, enter a title under **Panel title** or toggle the **Show panel title** to hide the title.
```

4. Choose **Save**.

> Changing panel titles only affects the particular panel on the particular dashboard, and won't affect any other panel containing that same visualization or any other dashboard.

Reviewer 4: ```suggestion
Changing panel titles only affects the particular panel on the particular dashboard and won't affect any other panel containing that same visualization or any other dashboard.
```

{: .note}

## Arranging panels

To organize panels, arrange them side-by-side, or resize them, you can use these options:

> - To move, select and hold the panel title or the panel top and drag to the new location.

Reviewer 4: ```suggestion
- To move a panel, select and hold the panel title or the panel top and drag to the new location.
```

Reviewer 4: Is there better terminology than "panel top"?

> - To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.

Reviewer 4: ```suggestion
- To resize a panel, choose the resize icon in the panel's lower-right corner and drag to the new dimensions.
```

> - To view in fullscreen mode, select the gear icon (edit mode) or vertical ellipsis (⋮) in the top right of the panel and select **Maximize panel**. To minimize the fullscreen mode, select the gear icon or vertical ellipsis and select **Minimize**.

Reviewer 4: ```suggestion
- To view a panel in full screen mode, choose the gear icon (edit mode) or vertical ellipsis (⋮) at the top right of the panel and select **Maximize panel**. To minimize the full screen mode, choose the gear icon or vertical ellipsis and select **Minimize**.
```

> The following is an example of a customized dashboard created in this tutorial.

Reviewer 4: ```suggestion
The following is an example of a customized dashboard created by using this tutorial.
```

---
layout: default
title: Creating dashboards
nav_order: 30
has_children: true
---

# Creating dashboards

> A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application you can:

Reviewer 5: ```suggestion
A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application, you can:
```

> A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application, you can:

Reviewer 3: Technically, dashboards are not just limited to displaying visualizations (they can render almost anything in a panel), but this is fine as an approximation.

Author: Revised to reflect Dashboards broader capabilities.

> - Get started with pre-built panels and dashboards and customize display properties to create the dashboard you need.

Reviewer 3: It seems like this list should either describe the general capabilities of dashboards (what you can do, which would more closely map to the sections of the tutorial, or describe the utility of dashboards (why it's useful). This first item seems like neither. Maybe something along the lines of

```suggestion
- Create a collection of visualizations that can be filtered and customized
```

> - Analyze, monitor, and alarm on metrics, logs, and traces across multiple data sources.

Reviewer 3: This seems out of scope for a beginner-targeted tutorial, particularly because we haven't yet launched the anywhere project. I'd recommend removing.

Author: Good point. In thinking more braodly about dashboards capabilitites, I've revised this intro.

- Create interactive dashboards and share them with anyone in your organization.

<img src="{{site.url}}{{site.baseurl}}/images/dashboards/dashboard-index.png" alt="Single-view dashboard in OpenSearch Dashboards" width="600" height="300">

> Before you begin this tutorial, make sure you've installed OpenSearch and OpenSearch Dashboards and that you've connected to Dashboards at [http://localhost:5601](http://localhost:5601). The username and password are `admin`.

Reviewer 3: A bit out of scope, but for the target user, I think it's a better assumption to expect that they'll either be doing it on their existing installation of OpenSearch Dashboards, or else on the playground: https://playground.opensearch.org/app/home.  Although neither is perfect, since sample data may not be available on the former, and the latter doesn't support saving.

Author: @joshuarrrr Please confirm the revised text addresses your feedback.

{: .note}

## Getting familiar with user interface

In this tutorial you'll learn the basics of creating a dashboard using the **Dashboard** application and OpenSearch sample data. Before getting started, get familiar with the Dashboard user interface. The user interface comprises five main parts, as shown in the following image:

<img src="{{site.url}}{{site.baseurl}}/images/dashboards/user-interface.png" alt="Dashboard user interface" width="600" height="300">

- The **navigation panel** (A) on the left contains the OpenSearch Dashboards applications.
- The **toolbar** (B) contains frequently used commands and shortcuts.
- The **search** bar (C) allows you to search for documents and other objects and add filters.

> - The **time filter** (D) allows you to customize the time and date.

Reviewer 3: If we're having separate items for search and time bar, I'd also include the filter bar (below C), or else include it as part of C.

- The **panel** (E) allows you to add existing visualizations to the dashboard or create new ones for the dashboard.

> 

Reviewer 3: nit - extra empty line.

_Panel_ is a term used to refer to a visualization displayed on a dashboard. The terms _panel_ and _visualization_ may be used interchangeably throughout this and other Dashboard-related documentation.
{: .note}

## Adding sample data

> If you haven't already added the OpenSearch Dashboards sample datasets as described in [Quickstart guide for OpenSearch Dashboards](https://opensearch.org/docs/latest/dashboards/quickstart-dashboards/), add them before proceeding to the next steps. **Sample eCommerce orders** is used for this tutorial. 

Reviewer 3: Out of scope - we should create an issue about making re-usable snippets for repeated content like this. As we add more pages, it will be increasingly difficult to keep all these copy/pasted sections up to date and in sync.

Author: Good point. Deleted.

To add the sample dataset:

1. On the OpenSearch Dashboards **Home** page, choose **Add sample data**.
2. Choose **Add data** and choose **Sample eCommerce orders**.

## Creating dashboards

> Now that you've added the sample data, you can create a dashboard using that data. The sample dataset has existing sample visualizations, and you can use those visualizations or create new visualizations for the dashboard. For this tutorial, you'll do both.

Reviewer 3: I think we could also have them look at one of the sample dashboards first, and then say the goal of this tutorial will be to create something like that from scratch.

Author: Good point. Change made.

1. From the navigation panel, choose **Dashboard**.
2. From the **Dashboards** window, choose **Create Dashboard**.
3. Set the time filter to **Last 30 days**.

> 4. Choose **Add an existing**.

Reviewer 3: This link isn't easy to spot in the entire page

```suggestion
4. From the panel choose **Add an existing**.
```

5. From the **Add panels** window, choose **Create new** and **Visualization**. This is a shortcut for creating a dashboard and a visualization in one flow. Alternatively, you can use the **Visualize** application to create visualizations and then add them to the dashboard.

> 6. From the **New Visualization** window, choose **[eCommerce] Average Sold Quantity**. The donut chart is added to the dashboard.

Reviewer 3: These steps don't seem to agree. In this flow we're just adding an existing visualization, so we should choose (click) `[eCommerce] Average Sold Quantity` from the list (or search for it and click). Are we just wanting them to notice that the `Create new` in this flyout is an alternative to the `Create new` button they'll actually use in the next section?

Author: Thanks for this input. I was mixing the two tasks - adding an existing viz and creating a new viz. Revised this and the following section.

Author: Revised.

You've created a basic dashboard like the following, which you'll continue using throughout this tutorial.

![Creating a basic dashboard]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-basic.png)

## Creating and saving visualizations

Continuing with the dashboard you created in the preceding steps, you'll create a new visualization and save it to the dashboard.

1. From the dashboard toolbar, choose **Create new**.
1. From the **New Visualization** window, choose **Vertical Bar** and then select the index pattern **opensearch_dashboards_sample_data_ecommerce**. You'll see an image like the following.

    ![New visualization vertical bar chart]({{site.url}}{{site.baseurl}}/images/dashboards/new-bar-viz.png)

1. Configure the chart's **Data** settings by verifying **Metrics** is **Y-axis Count**.
1. From the **Buckets** panel, choose **Add** and select **X-axis** from the **Add Bucket** dropdown list.
1. From the **Aggregation** dropdown list, select **Date histogram**.
1. Choose **Update**. You'll see a bar chart visualization like the following:

    ![Configuring a visualization]({{site.url}}{{site.baseurl}}/images/dashboards/configure-bar-viz.png)

> 1. From the toolbar, select **Save**

Reviewer 3: These steps are a bit of a tangent from actually learning about dashboards. For the purpose of the tutorial, it might be better to just save the default vis, without any configurations, so they see the flow of dashboard->vis->dashboard with out extraneous steps.

Author: Revised.

Author: Revised.

1. In the **Save visualization** window, add a title for the visualization. For example, the title for the bar chart panel is Order Date.
1. Select **Save and return**.  

> The bar chart visualization is now saved and you are taken back to the dashboard. You'll see three visualizations on the dashboard, as shown in the following image: 

Reviewer 3: The screenshot only shows 2?
```suggestion
The bar chart visualization is now saved and you are taken back to the dashboard. You'll see two visualizations on the dashboard, as shown in the following image: 
```

![Dashboard showing visualizations combined in a single view]({{site.url}}{{site.baseurl}}/images/dashboards/new-dashboard.png)

> ## Adding panels

Reviewer 3: maybe?
```suggestion
## Adding subsequent panels
```

otherwise we can just distinguish that using the `Add` option from the toolbar is the more common way to add panels, because the method in the "Creating dashboards" section is _only_ for the case of a completely empty dashboard.

Continuing with the dashboard you created in the preceding steps, you'll add a visualization to the dashboard.

1. From the dashboard toolbar, choose **Add**.
1. From the **Add panels** window, choose **[eCommerce] Sales by Category**.
1. Select `x` to close the **Add panels** window. 

You'll see an area chart visualization display on the dashboard, as shown in the following image: 

![Adding another panel to the dashboard]({{site.url}}{{site.baseurl}}/images/dashboards/new-area-viz.png)

> ## Editing visualizations

Reviewer 3: Should we save the dashboard at this point? We've gone a decent way in an unsaved state, and now might be the time to explain the difference between edit and view modes.

Author: Agreed. Revised.

> You can edit visualizations, configure panels, and specify labels for a panel. For example, you can change the visualization's colors, title, and other attributes. Visualizations can be edited in the panel itself or in edit mode. Only changes made in edit mode are persistent, that is, they are saved in the visualization itself and will be reflected on any dashboard that uses the visualization. Editing in the panel is helpful for making minor changes at the dashboard level without saving them.

Reviewer 3: I think usage of the term "edit mode" here is confusing, particularly because the dashboard itself has an edit mode and a view mode. What we want to distinguish between is tweaking the panel itself vs actually editing the visualization itself. Panel changes include:

1. Toggling visibility of the legend
2. Changing colors via the legend
3. Toggling visibility of the panel title
4. Changing the panel title

I think it might be easier to re-arrange this section to just cover those topics as part of the tutorial, and then mention that it's separate from editing the visualization.

Author: Revised.

Author: @joshuarrrr Will you review this revised section? Do you think we should have a section herein on editing visualizations? Or do you think that tutorial would be better suited to the Building data visualizations section?

Follow these steps to edit in the panel:

1. Select a category in the legend and then select a color from the flyout.  

Follow these steps to use edit mode:

1. Select the gear icon on the area chart panel.
2. From the **Options** window, select **Edit visualization**.
3. From the visualization legend, select a category and then select a color from the flyout. The area chart updates with your edit.
4. Select **Save and return**. 

Using the **Options** window also allows you to edit or hide the title for a particular panel without applying the changes across the dashboard overall.

1. Select the gear icon on the panel.
2. From the **Options** window, select **Edit panel title**.
3. From the **Customize panel**, add a title under **Panel title** or toggle the **Show panel title** to hide the title.
4. Choose **Save**.

To hide the legend, select the list icon in the panel's lower left corner.

## Arranging panels

To organize panels, arrange them side-by-side, or resize them, you can use these options:

- To move, select and hold the panel title and drag to the new location.
- To resize, select the resize icon in the panel's lower right corner and drag to the new dimensions.

> - To view in fullscreen mode, select the gear icon to open the panel menu and then then select **More > Maximize panel**.

Reviewer 3: I suggest adding a step to mention Minimize as well, so they can get back to the previous state.

Author: Revised.

Author: Added.

The following is an example of a customized dashboard showing visualizations arranged side by side and without legends:

![Dashboard with panels arranged side by side and without legends]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-arranged.png)

## Filtering documents

> Filtering documents is helpful when you want to review data in an ad hoc manner. You can filter documents by applying filters and/or using Dashboards Query Language (DQL). For information about DQL, see [Using Dashboards Query Language]({{site.url}}{{site.baseurl}}/dashboards/discover/dql/). The dashboard queries the data it needs using the filter and/or DQL and applies the filter across the dashboard, that is, to all visualizations on the dashboard.  

Reviewer 3: I'd drop the first sentence. DQL-based searches and filters feel more complex than the rest of this tutorial.

To filter the data by applying a filter:

1. From the dashboard toolbar, choose **Add filter**.
2. From the **Edit filter** flyout, select a field, operator, and value. For example, select **geoip.region_name**, **is**, and **California**, as shown in the following image:

    ![Adding a filter]({{site.url}}{{site.baseurl}}/images/dashboards/edit-filter.png)    

3. Choose **Save**. The filter is applied to all visualizations on the dashboard.
4. To save the dashboard and filter, choose **Save** from the toolbar and then add a title and choose **Save** from the **Save dashboard** window.

![Saving the dashboard and filter]({{site.url}}{{site.baseurl}}/images/dashboards/save-dashboard.png)

To filter documents using DQL:

1. From the DQL toolbar, specify the search term. For example, input `geoip.region_name: New York` in the query field.
2. Select **Update**. A dashboard similar to the following is displayed:

> ![Filtering using DQL]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-dql.png)

Reviewer 3: I'd recommend cutting and saving for a more advanced tutorial.

Author: Agreed. Removed to include in a separate tutorial.

Author: Agreed

---
layout: default
title: Creating dashboards
nav_order: 30
has_children: true
---

# Creating dashboards

> A dashboard combines multiple data visualizations into a single view. With the **Dashboard** application you can:

Author: Revised
